src/components/main-view/main-view.jsx

```diff
@@ -16,107 +20,138 @@ export class MainView extends React.Component {
-      register: null,
     };
+  }
 
 
-  this.goBack = this.goBack.bind(this);
-  this.onLoggedIn = this.onLoggedIn.bind(this);
+  componentDidMount() {
+    let accessToken = localStorage.getItem("token");
+    if (accessToken !== null) {
+      this.setState({
+        user: localStorage.getItem("user"),
+      });
+      this.getMovies(accessToken);
+    }
   }
 
-  componentDidMount() {
+
+  getMovies(token) {
     axios
-      .get("https://movie-api-1684.herokuapp.com/movies")
+      .get("https://movie-api-1684.herokuapp.com/movies", {
+        headers: { Authorization: `Bearer ${token}` },
+      })
       .then((response) => {
-        // Assign the result to the state
-        this.setState({ movies: response.data });
+        this.setState({
+          movies: response.data,
+        });
       })
       .catch(function (error) {
         console.log(error);
       });
   }
 
-  onMovieClick(movie) {
-    this.setState({ selectedMovie: movie });
-  }
-  goBack() {
-    this.setState({ selectedMovie: null });
-  }
-  onLoggedIn(user) {
-    this.setState({ user });
-  }
-  startRegister() {
-    this.setState({ register : true });
-  }
-  endRegister() {
-    this.setState({ register : null });
+  
+  // props.onLoggedIn(data) - LoginView
+  onLoggedIn(authData) {
+    console.log(authData);
+    this.setState({
+      user: authData.user.Username,
+    });
+
+    localStorage.setItem("token", authData.token);
+    localStorage.setItem("user", authData.user.Username);
+    this.getMovies(authData.token);
   }
 
-  render() {
-    const { movies, selectedMovie, user, register } = this.state;
+  onLoggedOut() {
+		localStorage.removeItem('token');
+		localStorage.removeItem('user');
+		this.setState({
+			user: null,
+		});
+	}
 
-   if (register) {
-      return <RegistrationView endRegister={() => this.endRegister()} />;
-    } 
-
-    if (!user)
-      return  <LoginView startRegister={() => this.startRegister()}
-      onLoggedIn={(user) => this.onLoggedIn(user)}/>
+  render() {
+    const { movies, user } = this.state;
 
     // Before the movies have been loaded
     if (!movies) return <div className="main-view" />;
 
     return (
-      <div className="main-view">
-      <Container>
-      <Row className="justify-content-md-center">
-        {selectedMovie ? (          
-            <Col md={8}>
-          <MovieView movie={selectedMovie} goBack={() => this.goBack()} />
-          </Col>
-        
-        ) : (
-          movies.map((movie) => (
-            <MovieCard
-              key={movie._id}
-              movie={movie}
-              onClick={(movie) => this.onMovieClick(movie)}
-            />
-          ))
-        )}
-        </Row>
-      </Container>
+      <Router>
+        <div className="main-view">
+          <h2>Main View</h2>
+
+        <ul>
+          <li>
+          <Route path="/register" render={() => <RegistrationView />} />
+          <Link to="/register">Register</Link>
+          </li>
```

Comment:
I think you mixed things up here with React-Router `Link` and `Route`.
With this code you are now rendering your login View within a listitem`<li>`.

I don't think you want that, right?
`<Route path="/register" render={() => <RegistrationView />} />`
This code will render the RegistrationView at this current spot in your code when `/register` is in your url

`<Link to="/register">Register</Link>` on the other hand creates a link similar as a hyperlink in a navbar but a bit different. With this you can call the '/register' url and the RegisterView is rendered.

So you need to separate the Routes fro the links here like so:
**navbar**
```
<nav>
    <ul>
        <li>
            <Link to="/***">xxx</Link>
        </li>
        <li>
            <Link to="/***">xxx</Link>
        </li>
        <li>
            <Link to="/***">xxx</Link>
        </li>
    </ul>
</nav>
```
**Rendering your views**
<Route>
    <nav>... see above ...</nav>
        <Route exact path="***" render={xxx} />
        <Route path="***" render={xxx} />
        <Route path="***" render={xxx} />
<Route>

Reply:
Regarding `<Route path="/register" render={() => <RegistrationView />} />` is from the chapter, so i thought is right code. But not in this case? 

After your instruction now is like this.

  render() {
    const { movies, user } = this.state;

    // Before the movies have been loaded
    if (!movies) return <div className="main-view" />;

    return (
      <Router>
      <Navbar>
  
       <ul>
          <li>
          <Link to="/register">Register</Link>
          </li>
                
        <li>
        <Link to="/movies/:movieId">Movies</Link>
        </li>

        <li>
        <Link to="/">All List</Link>
        </li>

        <li>
        <Link to="/directors/:name">Directors</Link>
        </li>

        <li>
        <Link to="/genres/:name">Genres</Link>
        </li>
      </ul>
      </Navbar>

       {/* <Route path="/register" render={() => <RegistrationView />} />    */} **** the issue you mentioned earlier

          <Route exact path="/movies/:movieId" render={({match}) => 
          <MovieView movie={movies.find(m => m._id === match.params.movieId)}/>}/>
    
          <Route exact path="/"  render={() => 
            {if (!user) return (
            <LoginView onLoggedIn={(user) => this.onLoggedIn(user)}/> );
            return movies.map((m) => 
            <MovieCard key={m._id} movie={m} />);
            }}/>
           
  
         <Route exact path="/genres/:name" render={({match}) => {
             if (!movies) return <div className="main-view" />;
             return <GenreView genre={movies.find((m) => 
            m.Genre.Name === match.params.name).Genre}/>;
          }}/>
     

        <Route exact path="/directors/:name" render={({match}) => {
              if (!movies) return <div className="main-view" />;
              return <DirectorView genre={movies.find((m) => 
              m.Director.Name === match.params.name).Director}/>;
          }}/>

    </Router>
    );
  }
}

Reply:
What I meant is following
 You don't want to render a RegisterView or LoginView within a <li> lisitem. You need to separate the <Link> from the <Route>
 
 **Render links to views**
 ```
 <nav>
    <ul>
        <li>
            <Link to="/***">xxx</Link>
        </li>
        <li>
            <Link to="/***">xxx</Link>
        </li>
        <li>
            <Link to="/***">xxx</Link>
        </li>
    </ul>
</nav>
```

**Rendering your views**
```
    <Route exact path="***" render={xxx} />
    <Route path="/register" render={() => <RegistrationView />} /> 
    <Route path="/register" render={() => <LoginView />} /> 
    <Route ... /> 
```

Reply:
![LoginView](https://user-images.githubusercontent.com/71074389/104931010-25e2dd00-59a6-11eb-8fea-bea1c4d2a999.JPG)

After I click LoginView appear that screenshot and error **no such user**

So far I'm using `localhost:1234 `to access this. Or is this wrong?